<launch>

  <param name="/use_simulation" type="bool" value="true" />

  <arg name="world_name" default="$(find testbot)/world/dipole.world"/>

  <!-- these are the arguments you can pass this launch file, for example paused:=true -->
  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui" default="true"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>

  <!-- Vehicle pose -->
  <arg name="x" default="1.7"/>
  <arg name="y" default="0.0"/>
  <arg name="z" default="0.5"/>
  <arg name="roll" default="0.0"/>
  <arg name="pitch" default="0.0"/>
  <arg name="yaw" default="0.0"/>

  <!-- We resume the logic in empty_world.launch -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(arg world_name)"/>
    <arg name="debug" value="$(arg debug)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="headless" value="$(arg headless)"/>
  </include>

  <!-- urdf xml robot description loaded on the Parameter Server-->
  <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find testbot)/urdf/testbot.xacro'" />


  <!-- Run a python script to the send a service call to gazebo_ros to spawn a URDF robot -->
  <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
    args="-urdf -model testbot -param robot_description
          -x $(arg x) -y $(arg y) -z $(arg z)
          -R $(arg roll) -P $(arg pitch) -Y $(arg yaw)"/>

  <node name="complementary_filter" pkg="testbot" type="complementary_filter.py" />

  <node name="controller" pkg="testbot" type="controller_pid_with_anti_windup.py" output="screen"/>

  <!--<include file="$(find testbot)/launch/control_testbot.launch" />

  <node name="delta_control" pkg="pid" type="controller" >
      <param name="Kp" value="5.0" />
      <param name="Ki" value="0.0" />
      <param name="Kd" value="0.0" />
      <param name="upper_limit" value="10" />
      <param name="lower_limit" value="-10" />
      <param name="windup_limit" value="10" />
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" />
	  <remap from="/control_effort" to="/testbot/delta1"/>
	  <remap from="/state" to="/roll"/>
   </node> -->

</launch>
